### ISSUE TITLE
problem of recalculating the anchors

### ISSUE THREAD

**Issue by anguoyang** [AlexeyAB/darknet#688]
@AlexeyAB according to 
https://github.com/AlexeyAB/darknet#how-to-improve-object-detection
I used 
darknet.exe detector calc_anchors data/obj.data -num_of_clusters 9 -width 416 -height 416 to generate anchors, it was much different with the original one, this is my result:
anchors = 0.2659,0.1707, 74.9843,87.2402, 63.4272,132.3551, 96.7205,119.8974, 72.7019,179.2829, 137.8497,111.0177, 115.6360,152.7607, 113.0501,206.1598, 151.3696,166.3764
is it really correct? there is no error on training, but I am worry about the accuracy

**Comment by AlexeyAB** [comment]
Yes, calculation of anchors a little bit difference than used in the original Yolo.

> darknet.exe detector calc_anchors **data/obj.data** -num_of_clusters 9 -width 416 -height 416 to generate anchors, it was much different with the original one, this is my result:
> anchors = 0.2659,0.1707, 74.9843,87.2402, 63.4272,132.3551, 96.7205,119.8974, 72.7019,179.2829, 137.8497,111.0177, 115.6360,152.7607, 113.0501,206.1598, 151.3696,166.3764

Did you calculate it for PascalVOC dataset or for your dataset?

**Comment by anguoyang** [comment]
for my dataset

**Comment by AlexeyAB** [comment]
> it was much different with the original one

Because your Dataset much different with the Pascal VOC dataset )

As I see you have very small objects that will be smaller than 1x1 pixel when your image will be resized to 416x416 (network size).

**Comment by anguoyang** [comment]
@AlexeyAB yes, because we need to detect objects 80cm above, so we take pictures with similar distance, so the objects looks small

**Comment by anguoyang** [comment]
do I need to crop the object area, for example 640x480 ? the whole image is large( more than 4000x4000)

**Comment by anguoyang** [comment]
currently I resized the whole images(4032x3024) into 640x480 , then the smallest object is about 70x70 pixels, I suppose that when resized into 416x416, the smallest object is about 50x50 pixels

**Comment by AlexeyAB** [comment]
50x50 pixels objects can be detected easily, so all is ok.

**Comment by anguoyang** [comment]
it seems that the result is not so good. I have compared these 2 scenario(both on 10 classes):
1. 8 images for each class, training with yolo v2-densenet, object is big in the image
result: over-fitting but not so bad
2. 500 images for each class, training with yolo v3, some objects are very small in the image
result: very bad
I will try to crop the objects area out of the whole image and resize it into 640x480, and then training
